src/service/variable-source.js

```diff
@@ -121,6 +127,44 @@ export class VariableSource {
 
     /** @private {boolean} */
     this.initialized_ = false;
+
+    /** 
+     * The whitelist of variables allowed for variable substitution.
+     * @private @const {?Array<string>} 
+     */
+    this.whitelist_ = this.getVariableWhitelist_();
+  }
+
+  /**
+   * @return {?Array<string>} The whitelist of allowed AMP variables. (if provided in 
+   *     a meta tag).
+   * @private
+   */
+  getVariableWhitelist_() {
+    if (this.whitelist_) {
+      return this.whitelist_;
+    }
+
+    if (!this.ampdoc) {
```

Comment:
Add a TODO to remove this after fixing the tests?

Reply:
Done. I'll also create an issue when I create a pull request to upstream.